provider/datagen/src/transform/cldr/time_zones/convert.rs

```diff
@@ -55,6 +58,37 @@ pub(crate) fn compute_bcp47_tzids_btreemap(
     bcp47_tzids
 }
 
+/// Returns a map from BCP-47 ID to a single canonical long identifier.
+///
+/// For example: "inccu" to "Asia/Kolkata"
+pub(crate) fn compute_canonical_tzids_btreemap(
+    bcp47_tzids_resource: &BTreeMap<TimeZoneBcp47Id, Bcp47TzidAliasData>,
+) -> BTreeMap<TimeZoneBcp47Id, String> {
+    let mut canonical_tzids = BTreeMap::new();
+    for (bcp47_tzid, bcp47_tzid_data) in bcp47_tzids_resource.iter() {
+        if Some(true) == bcp47_tzid_data.deprecated {
+            // skip
+        } else if let Some(iana) = &bcp47_tzid_data.iana {
+            canonical_tzids.insert(*bcp47_tzid, iana.clone());
+        } else if let Some(iana) = &bcp47_tzid_data
+            .alias
+            .as_ref()
+            .and_then(|s| s.split(' ').next())
+        {
+            canonical_tzids.insert(*bcp47_tzid, String::from(*iana));
```

Comment:
@srl295: can you confirm that this algorithm is correct? First check for the iana attribute, and if it's not there, then choose the first entry in the aliases list.

Reply:
I'm not that familiar with the algorithm… i can take a look

Reply:
If you want _IANA_'s preferred ID, your description above sounds right.

But then you should get `Asia/Kolkata` because that's what the data has. 

`<type name="inccu" description="Kolkata, India" alias="Asia/Calcutta Asia/Kolkata" iana="Asia/Kolkata"/>` (just as in the spec).

Reply:
> can you confirm that this algorithm is correct? First check for the iana attribute, and if it's not there, then choose the first entry in the aliases list.

Yes, this is correct.